docs/astro/sql/operators/load_file.rst

```diff
@@ -0,0 +1,162 @@
+
+    .. literalinclude:: ../../../../example_dags/example_load_file.py
+       :language: python
+       :start-after: [load_file_example_4_start]
+       :end-before: [load_file_example_4_end]
+
+#. **output_table** - We can specify the output table to be created by passing in this parameter, which is expected to be an instance of ``astro.sql.table.Table``. Users can specify the schema of tables by passing in the ``columns`` parameter of ``astro.sql.table.Table`` object, which is expected to be a list of the instance of ``sqlalchemy.Column``. If the user doesn't specify the schema the schema is inferred using pandas.
+
+    .. literalinclude:: ../../../../example_dags/example_load_file.py
+       :language: python
+       :start-after: [load_file_example_5_start]
+       :end-before: [load_file_example_5_end]
+
+
+Parameters to use when loading a file to pandas dataframe
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#. ``columns_names_capitalization`` Control col names case of the dataframe generated from the file. The default value is ``original``.
+        *  **original** - Remains the same as the input file
+        *  **upper** - Convert to uppercase
+        *  **lower** - Convert to lowercase
+
+    .. literalinclude:: ../../../../example_dags/example_load_file.py
+       :language: python
+       :start-after: [load_file_example_6_start]
+       :end-before: [load_file_example_6_end]
+
+How load_file Works
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. figure:: ../../../images/defaultPath.png
+
+
+When we load a file located in cloud storage to a cloud database, internally the steps involved are:
+
+Steps:
+
+#. Get the file data in chunks from file storage to the worker node.
+#. Send data to the cloud database from the worker node.
+
+This is the default way of loading data into a table. There are performance bottlenecks because of limitations of memory, processing power, and internet bandwidth of worker node.
+
+
+Improving bottlenecks by using native paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. figure:: ../../../images/nativePath.png
+
+Some of the cloud databases like Bigquery and Snowflake support native paths (complete list of supported native paths :ref:`supported_native_path`) to ingest data from cloud storage directly. Using this we can ingest data much quicker and without any involvement of the worker node.
+
+Steps:
+
+#. Request destination database to ingest data from the file source.
+#. Database request file source for data.
+
+This is a faster way to ingest data as there is only one network call involved and usually the bandwidth between vendors is high. Also, there is no requirement for memory/processing power of the worker node, since data never gets on the node. There is significant performance improvement due to native paths as evident from benchmarking results.
```

Comment:
This may not be the case.

Depending on the native load implementation, we have an "expensive" setup (e.g. creating Google Storage Transfer configuration or creating Snowflake remote staging) - which means that for small datasets, the native load may be slower when compared to using the Pandas dataframe approach.

Reply:
Updated the doc to reflect this fact.